crates/ruff_python_formatter/src/expression/expr_await.rs

```diff
@@ -20,7 +20,7 @@ impl FormatNodeRule<ExprAwait> for FormatExprAwait {
             [
                 token("await"),
                 space(),
-                maybe_parenthesize_expression(value, item, Parenthesize::IfRequired)
+                maybe_parenthesize_expression(value, item, Parenthesize::IfBreaks)
```

Comment:
Should this be `IfBreaksOrIfRequired`?

Reply:
That one would regress `asyncio.gather` 
```diff
 # Long lines
 async def main():
-    await asyncio.gather(
-        asyncio.sleep(1),
-        asyncio.sleep(1),
-        asyncio.sleep(1),
-        asyncio.sleep(1),
-        asyncio.sleep(1),
-        asyncio.sleep(1),
-        asyncio.sleep(1),
+    await (
+        asyncio.gather(
+            asyncio.sleep(1),
+            asyncio.sleep(1),
+            asyncio.sleep(1),
+            asyncio.sleep(1),
+            asyncio.sleep(1),
+            asyncio.sleep(1),
+            asyncio.sleep(1),
+        )
     )
```
(i'm happy too implement a different solution btw, this was just the solution that i found to work best by trial and error)

Reply:
Ahh yeah, I agree that that's worse. I feel like I need to have a better understanding of why these `Parenthesize` values are leading to these changes before I'd be comfortable approving. Why does `IfBreaks` break that way? Will using `IfBreaks` ever lead to syntax errors?

Reply:
No, that's handled by `OptionalParentheses::Multiline`

Reply:
I'll make some better examples why we get this behavior

Reply:
Thank you, I'm trying to make sure I play Micha and really understand the impact of formatting changes before approving.

Reply:
From the perspective of the await, it needs `OptionalParentheses::Never` because the child is already parenthesized, and we're preferring those parentheses over adding them around the await. For the await's child, we have four options:

* Optional: Add parentheses if the child breaks, which we want e.g. for `asyncio.gather`. Keeps the child's parentheses from the source code, which we want to remove.
* IfBreaks: Add parentheses if the child breaks, which we want e.g. for `asyncio.gather`. Discards child parentheses, which we want.
* IfRequired: Discards child parentheses, which we want. Also discards them when the child breaks but has its own inner parentheses, which would lead to e.g.
    ```python
    await asyncio.gather(
        fut1,
        fut2,
    )
    ```
    , which don't want.
* IfBreaksOrIfRequired: Special case for return type annotations.
 
I added comments to the parentheses types to make them better understandable in the future